### PR TITLE
Fix empty operation names with Distributed Tracing V2 (#3156)

### LINKED ISSUE #3156: Empty operation names with Distributed Tracing V2
Customers in GitHub issue #2662 have reported empty operation names in Application Insights when using Distributed Tracing V2 with .NET Isolated apps.

Example from Application Insights:

Image

### PR DESCRIPTION
**Pull request by YunchuWang**
# Summary
## What changed?
- Populate an empty Application Insights `operation_Name` from the existing `OperationTelemetry.Name`.
- Preserve the existing opt-in orchestration instance ID suffix behavior.
- Add focused unit coverage and a real Distributed Tracing V2 orchestration/activity E2E regression.
- Add a release note for the user-visible bug fix.

## Why is this change needed?
- The V2 Durable Task telemetry exporter sets the request/dependency telemetry `Name`, but leaves `Context.Operation.Name` empty.
- The existing initializer only repaired that field when `IncludeInstanceIdInOperationName` was enabled, even though the option defaults to `false`.
- As a result, default V2 telemetry could appear under an empty operation name in Application Insights.

## Issues / work items
- Resolves #3156
- Related #3266

---

# Project checklist
- [x] Documentation changes are not required
  - [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
- [ ] Release notes are not required for the next release
  - [x] Otherwise: Notes added to `release_notes.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] No extra work is required to be leveraged by OutOfProc SDKs
  - [ ] Otherwise: Work tracked here: #issue_or_pr_in_each_sdk
- [x] No change to the version of the `WebJobs.Extensions.DurableTask` package
  - [ ] Otherwise: Major/minor updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
- [x] No EventIds were added to `EventSource` logs
  - [ ] Otherwise: Ensure EventIds are within the supported range in the existing Windows infrastructure (validate via deployed telemetry). If needed, extend the range via a PR such as https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files
- [ ] This change should be added to the `v2.x` branch
  - [x] Otherwise: This change applies exclusively to `WebJobs.Extensions.DurableTask` v3.x and will be retained only in the `dev` and `main` branches
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:

---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot CLI (GPT-5.6 Sol)
- AI-assisted areas/files: telemetry initializer, unit regression, correlation E2E regression, and release note
- What you changed after AI output: The initial `Activity.Current`-based implementation was rejected after the full E2E suite exposed missing ambient activity state. The final revision uses the telemetry's explicit `Name` value and was reviewed against the complete diff and local evidence.

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing
## Automated tests
- Result: Passed
  - Focused unit + V2 E2E regressions: 2/2
  - Non-operation telemetry regression: 1/1
  - `DurableTaskInstanceIdTelemetryInitializerTests`: 9/9 on .NET 8 and 9/9 on .NET 10
  - `CorrelationEndToEndTests`: 29/29 on .NET 8
  - New V2 E2E regression: 1/1 on .NET 10

## Manual validation (only if runtime/behavior changed)
- Environment (OS, .NET version, components): Windows, .NET 8/.NET 10, local Functions V2 test host, Azurite
- Steps + observed results:
  1. Ran the new V2 orchestration/activity E2E regression against the original implementation. The orchestration request, activity dependency, and activity request all had `Context.Operation.Name == null`.
  2. Applied the fix and reran the same path. Each operation name matched its existing telemetry name (`orchestration:SayHelloWithActivity` or `activity:Hello`).
  3. Ran the focused and full correlation test suites listed above.
  4. Deployed the PR build to a temporary Windows Consumption Function App with Application Insights and ran five `FunctionChaining` orchestrations. All five completed.
  5. Queried the ingested `requests` telemetry. The 25 requests were grouped under `activity:SayHelloActivity` (15), `orchestration:FunctionChaining` (5), and `DurableFunctionsPatterns_HttpStart` (5), with zero empty operation names.
- Evidence (optional): Local red/green and live Azure query screenshots were captured. The temporary Azure resource group was deleted after validation.

---

# Notes for reviewers
- Baseline operation-name population intentionally does not depend on `Activity.Current`; the full correlation suite demonstrated that ambient activity state is not always available when telemetry is initialized.
- Existing nonempty operation names are not overwritten, and instance ID suffixing remains opt-in.
- The upstream `Azure/durabletask` V2 exporter still omits `Context.Operation.Name`; this extension-side initializer fix provides immediate coverage without expanding this PR's scope.

### CODE REVIEW

File: src/WebJobs.Extensions.DurableTask/Correlation/DurableTaskInstanceIdTelemetryInitializer.cs [hunk 1]
```diff
@@ -52,12 +61,6 @@ public void Initialize(ITelemetry telemetry)
...
             {
-                // If the operation name is empty, use the activity display name
-                if (string.IsNullOrEmpty(telemetry.Context.Operation.Name))
-                {
-                    telemetry.Context.Operation.Name = activity.DisplayName;
-                }
-
                 // Append instance ID to operation name if not already present
                 if (!telemetry.Context.Operation.Name.Contains(instanceId))
                 {
```
[the remark this answers is not in the document]
Comment (YunchuWang) [reply src/WebJobs.Extensions.DurableTask/Correlation/DurableTaskInstanceIdTelemetryInitializer.cs:71]: Fixed in d8a7eda6. Restored the Activity.DisplayName fallback before Contains when non-OperationTelemetry has an empty operation name, preserving the prior behavior without changing the new OperationTelemetry.Name baseline path. Added an ExceptionTelemetry regression that reproduced the NullReferenceException before the fix; the initializer suite now passes 9/9 on both .NET 8 and .NET 10.

**Review by bachuv — APPROVED** [review]
